localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/parse.py

```diff
@@ -0,0 +1,19 @@
+import logging
+
+from localstack.http import Response
+
+from ..api import RestApiGatewayHandler, RestApiGatewayHandlerChain
+from ..context import RestApiInvocationContext
+
+LOG = logging.getLogger(__name__)
+
+
+class InvocationRequestParser(RestApiGatewayHandler):
```

Comment:
naming might be a bit misleading here: we probably think this handler will populate the context from the linked `deployment`, by "routing" the request depending on its path to the right APIGW `Resource` & `Method`. Naming is not set in stone and will be updated in follow up PRs.

Reply:
Interesting, this is exactly what I thought this class would do 😄 
But what is it intended to do then?

Reply:
I think this class might still be needed under this name, to parse for example the `RAW_URI` from the request and such things.
And we can also have a "router" handler that will properly match the incoming request to the right APIGW resource. Once we start splitting things, I think naming will become clearer. I still think we might need a parser somewhere 😄